src/main/kotlin/com/valensas/ftp/model/FTPClient.kt

```diff
@@ -2,15 +2,34 @@ package com.valensas.ftp.model
 
 import org.apache.commons.net.ftp.FTP
 import org.apache.commons.net.ftp.FTPClient
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
 import javax.naming.AuthenticationException
 
 open class FTPClient : FTPClient() {
-    open fun authAndConnect(connectionModel: ConnectionModel) {
-        this.connect(connectionModel.host, connectionModel.port)
-        if (!this.login(connectionModel.username, connectionModel.password)) {
-            throw AuthenticationException("Authentication failed")
+    private val logger: Logger = LoggerFactory.getLogger(javaClass)
+
+    private var _retryConnectionTimeouts: List<Int> = listOf(0)
+    var retryConnectionTimeouts: List<Int>
+        get() = _retryConnectionTimeouts
+        set(value) {
+            _retryConnectionTimeouts = listOf(0) + value
+        }
+
+    fun authAndConnect(connectionModel: ConnectionModel) {
+        run breaking@{
+            retryConnectionTimeouts.forEach {
+                try {
+                    this.defaultTimeout = it
+                    connectToServer(connectionModel)
+                    return@breaking
+                } catch (e: AuthenticationException) {
+                    throw e
```

Comment:
Bilmediğim için soruyorum Authentication Exception alırsa denemeye devam etmek istemiyor muyuz?

Reply:
Yok tam tersi aslında. Authentication error alınca durması için burada tekrar throw ediyorum. Diğer exceptionlarda retry sayısı kadar tekrar login etmeyi deniyor. @KaanSimsek19

Reply:
> Yok tam tersi aslında. Authentication error alınca durması için burada tekrar throw ediyorum. Diğer exceptionlarda retry sayısı kadar tekrar login etmeyi deniyor. @KaanSimsek19

tamamdır bunu sormuştum aslında.